lib/external-housing-available/external-housing-available.controllers.js

```diff
@@ -1,6 +1,172 @@
+      //If address doesnt exist or length is 0. 
+      if (!geocodes.results || geocodes.results.length === 0) {
+        try {
+          //Log it into the logger
+          await Log.request({
+            type: 'externalhousingaddress.submit.failed.no_results',
+            severity: 'Notice',
+            request: request,
+          });
+        } catch (e) {}
+        throw Boom.badData('No results for that address');
+      } 
+
+      const firstGeocode = geocodes.results[0];
+
+      let addressObject = postalAddressLib.extractAddress(firstGeocode);
+
+      if (!addressObject.addressNumber) {
+        try {
+          await Log.request({
+            type: 'externalhousingaddress.submit.failed.street_number',
+            severity: 'Notice',
+            request: request,
+          });
+        } catch (e) {}
+        throw Boom.badData('Address is missing a street number');
+      }
+
+      //Address without a county, we want to use locality instead.
+      if (!addressObject.hasOwnProperty('addressCounty')) {
+        addressObject.addressCounty = addressObject.addressLocality;
+      }
+      if (
+        addressObject.addressRegion !== process.env.ADDRESS_LIMIT_STATE ||
+        addressObject.addressCounty !== process.env.ADDRESS_LIMIT_COUNTY
+      ) {
+        try {
+          await Log.request({
+            type: 'externalhousingaddress.submit.failed.county',
+            severity: 'Notice',
+            request: request,
+          }); 
+        } catch (e) {} 
+        throw Boom.badData('Sorry, We are only accepting properties in ' + `${process.env.ADDRESS_LIMIT_COUNTY}, ${process.env.ADDRESS_LIMIT_STATE}`); 
+      }
+
+      //Extracting the address using function from geoCoordinates lib
+      let geoCoordObject = geoCoordinatesLib.extractLocation(firstGeocode);
+      
+      const addressHash = postalAddressLib.hashAddress(addressObject);
+      addressObject.hash = addressHash;
+
+      // External Housing Available Object 
+      let externalHousingAvailableObject = {};
+
+      const existingAddresses = await postalAddressLib.getAddressesByHash(addressHash);
+      if (existingAddresses.length > 0) {
+        //If this is an exisiting address
+        externalHousingAvailableObject.PropertyId = existingAddresses[0].PropertyId;
+      } else {
+        //Else, create a new propertyId for non exisiting address
+      }
+
+      const standardizedAddress = postalAddressLib.standardizeAddress(addressObject);
+      addressObject.address = standardizedAddress; 
+
+      //Title
+      if (request.payload.title && request.payload.title !== request.payload.address) {
+        externalHousingAvailableObject.title = request.payload.title;
+      } else {
+        externalHousingAvailableObject.title = standardizedAddress; 
+      }
+      //AuthorId
+      if (request.payload.AuthorId && request.auth.credentials.subjectId === process.env.SUPER_ADMIN) {
```

Comment:
this makes setting an AuthorId on the payload required. the check for the credentials to match SUPER_ADMIN is redundant because we already throw earlier in this method if they don't. It looks like for external reviews we just set the AuthorId to the account.id, so let's do that here as well for consistency.